src/server/auth/server/api_server.go

```diff
@@ -613,16 +596,20 @@ func (a *apiServer) WhoAmI(ctx context.Context, req *auth.WhoAmIRequest) (resp *
 	}
 
 	// Get TTL of user's token
+	// TODO(acohen4): give this value a second look
 	ttl := int64(-1) // value returned by etcd for keys w/ no lease (no TTL)
 	if callerInfo.Subject != auth.PpsUser {
 		token, err := auth.GetAuthToken(ctx)
 		if err != nil {
 			return nil, err
 		}
-		ttl, err = a.tokens.ReadOnly(ctx).TTL(auth.HashToken(token)) // lookup token TTL
+		ttl, err = a.lookupAuthTokenTTLSeconds(ctx, auth.HashToken(token))
 		if err != nil {
 			return nil, errors.Wrapf(err, "error looking up TTL for token")
 		}
+		if ttl < -1 {
```

Comment:
the semantics here are kind of strange. Let me know if you have a better idea to make this more clear.

a TTL of -1 is returned to the user to communicate "lives forever". This is to be consistent with the current behavior, and seems to make good sense. It's strange because if the TTL value is less than 0, it could also just be expired, so right now we're interpreting anything less than -1 as expired.

One way of writing this more clearly is to just return the ErrBadToken directly from `lookupAuthTokenTTLSeconds() `method